keck_vnc_launcher.py

```diff
@@ -123,10 +125,16 @@ def start(self):
         ##---------------------------------------------------------------------
         #todo: handle blank password error properly
         self.is_authenticated = False
-        if self.do_authenticate:
-            self.firewall_pass = getpass(f"Password for firewall authentication: ")
-            self.is_authenticated = self.authenticate(self.firewall_pass)
-            if not self.is_authenticated:
+        if self.do_authenticate == True:
+            self.firewall_pass = getpass.getpass(f"Password for firewall authentication: ")
+            try:
+                self.is_authenticated = self.authenticate(self.firewall_pass)
+            except:
```

Comment:
Take a look at handle_fatal_error which was designed to handle errors in a common way, including the stack trace dump to log

Reply:
These are not being handled as fatal errors: they're caught, logged, then the program continues on.